web_tool_script_1.R

```diff
@@ -26,15 +26,9 @@ source("0_global_functions.R")
 source("0_web_functions.R")
 source("0_json_functions.R")
 
-if (rstudioapi::isAvailable()) {
-  portfolio_name_ref_all <- c("TestPortfolio_Input")
-  working_location <- dirname(rstudioapi::getActiveDocumentContext()$path)
-  set_web_parameters(file_path = paste0(working_location,"/parameter_files/WebParameters_2dii.yml"))
-} else {
-  portfolio_name_ref_all = get_portfolio_name()
```

Comment:
I don't think we can remove this part, as this is where the script takes the information about the portfolio from, when called via command line. The webtool will mostly be called via command line in production, and get_portfolio_name() ensures it gets the correct information.
Same goes for Docker paramters instead of web parameters config. But happy to change it for the first part of the IF-statement

Reply:
Thanks for checking. 

Okay, if the code does what you expect when called interactively and from the terminal, then it may only need a new  `else` block to define the behavior when called from a non interactive session but not in the terminal -- like I did by knitting an rmarkdown document. I'm unsure here because I don't understand what `commandArgs()` does (my experiments didn't clarify help).

If the `if()` `else` code gets long you may abstract the details in something like this (here in pseudo code). The code should do the same but should be easier to understand at first glance:

```r
get_parameters <- function() {
  user_environment <- get_user_environment()
  switch(
    user_environment,
    "rstudio" = get_rstudio_params(),
    "docker" = get_docker_params(),
    "shell" = get_shell_params()
    stop("Unknown user environment")
  )
}
```